specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/compute.json

```diff
@@ -9462,17 +9474,424 @@
+          "description": "IP tag associated with the public IP. Example: SQL, Storage etc."
+        }
+      },
+      "description": "Contains the IP tag associated with the public IP address."
+    },
+    "VirtualMachinePublicIPAddressDnsSettingsConfiguration": {
+      "properties": {
+        "domainNameLabel": {
+          "type": "string",
+          "description": "The Domain name label. The concatenation of the domain name label and vm [what] will be the domain name labels of the PublicIPAddress resources that will be created"
+        }
+      },
+      "required": [
+        "domainNameLabel"
+      ],
+      "description": "Describes a virtual machines network configuration's DNS settings."
+    },
+    "VirtualMachinePublicIPAddressConfigurationProperties": {
+      "properties": {
+        "idleTimeoutInMinutes": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The idle timeout of the public IP address."
+        },
+        "deleteOption": {
+          "type": "string",
+          "description": "Specify what happens to the public IP address when the VM is deleted",
+          "default": "Delete",
+          "enum": [
+            "Delete",
+            "Detach"
+          ],
+          "x-ms-enum": {
+            "name": "DeleteOptions",
+            "modelAsString": false
+          }
+        },
+        "dnsSettings": {
+          "$ref": "#/definitions/VirtualMachinePublicIPAddressDnsSettingsConfiguration",
+          "description": "The dns settings to be applied on the publicIP addresses ."
+        },
+        "ipTags": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/VirtualMachineIpTag"
+          },
+          "description": "The list of IP tags associated with the public IP address."
+        },
+        "publicIPPrefix": {
+          "$ref": "#/definitions/SubResource",
+          "description": "The PublicIPPrefix from which to allocate publicIP addresses."
+        },
+        "publicIPAddressVersion": {
+          "type": "string",
+          "description": "Available from Api-Version 2019-07-01 onwards, it represents whether the specific ipconfiguration is IPv4 or IPv6. Default is taken as IPv4. Possible values are: 'IPv4' and 'IPv6'.",
+          "enum": [
+            "IPv4",
+            "IPv6"
+          ],
+          "x-ms-enum": {
+            "name": "IPVersions",
+            "modelAsString": true
+          }
+        },
+        "publicIPAllocationMethod": {
+          "type": "string",
+          "description": "Specify the public IP allocation type",
+          "enum": [
+            "Dynamic",
+            "Static"
+          ],
+          "x-ms-enum": {
+            "name": "PublicIPAllocationMethod",
+            "modelAsString": false
+          }
+        }
+      },
+      "description": "Describes a virtual machines IP Configuration's PublicIPAddress configuration"
+    },
+    "LoadBalancerAndPublicIPSku": {
```

Comment:
How is this named in the VMSS schema for external customer consumption?

Reply:
VirtualMachineScaleSetPublicIPAddressConfiguration doesn't have a sku in swagger

Reply:
The naming in the Netwokring Docs is PublicIPAddressSku, PublicIPAddressSkuName, PublicIPAddressSkuTier

This seems reasonable to replicate.

Can we confirm that is the naming in the NRP swagger spec.
If so I think we should go with that?



https://docs.microsoft.com/en-us/rest/api/virtualnetwork/publicipaddresses/createorupdate#publicipaddresssku